src/shogun/classifier/LDA.cpp

```diff
@@ -62,10 +60,6 @@ CLDA::~CLDA()
 bool CLDA::train_machine(CFeatures *data)
 {
 	REQUIRE(m_labels, "Labels for the given features are not specified!\n")
-	REQUIRE(
```

Comment:
You cannot just remove such a check, this will cause problems.
The reason why we have this check is that we cast the labels to binary later on.

E.g. there is a line
```
new CMulticlassLabels(static_cast<CBinaryLabels*>(m_labels))
```

Keep in mind `m_labels` here is of type `CLabels*`.

If you remove the check doing a static cast might cause memory errors. You will need to replace any static cast with a conversion, as in the commit I posted earlier.
Furthermore here, you could directly convert to multiclass labels, rather than going via the binary ones (though it is not clear whether that works yet, you will need to experiment)

Pls add a unit test where you put in labels of the wrong type and make sure that an exception is thrown (rather than the program crashing). This will also make sure that you understand what is going on

Reply:
I understand.. i am replacing static casts with conversions like in kernel svm. That line looks like ```new CMulticlassLabels(binary_labels(m_labels))``` the check can now be removed because we are already making sure of correct label type downstream (in binary_labels).

About directly converting to dense labels to multiclass that is not yet complete [here](https://github.com/shogun-toolbox/shogun/blob/32a6746251e4864269feb08702786ed1214be0b6/src/shogun/labels/MulticlassLabels.cpp#L196).